nix/pkgs.nix

```diff
@@ -11,11 +11,6 @@ final: prev: with final; {
 
   cabal = haskell-nix.cabal-install.${compiler-nix-name};
 
-  hlint = haskell-nix.tool compiler-nix-name "hlint" {
```

Comment:
removing `hlint` here means `nix develop` will provide the `hlint` version from the nixpkgs pin (currently hlint 3.3.6).

Reply:
(alternatively we could remove the check from nix `required` job and have the gihub check run `nix build '.#checks/hlint'`)